internal/tfr/getter_test.go

```diff
@@ -47,16 +47,60 @@ func TestGetDownloadURLFromHeaderWithPrefixedURL(t *testing.T) {
 func TestGetDownloadURLFromHeaderWithoutPrefixedURL(t *testing.T) {
 	t.Parallel()
 
-	testTerraformGet := "/terraform-aws-modules/terraform-aws-vpc"
-
-	testBaseURL := url.URL{
-		Scheme: "https",
-		Host:   "registry.terraform.io",
+	testCases := []struct {
+		name           string
+		moduleURL      url.URL
+		terraformGet   string
+		expectedResult string
+	}{
+		{
+			name: "BaseWithRoot",
+			moduleURL: url.URL{
+				Scheme: "https",
+				Host:   "registry.terraform.io",
+			},
+			terraformGet:   "/terraform-aws-modules/terraform-aws-vpc",
+			expectedResult: "https://registry.terraform.io/terraform-aws-modules/terraform-aws-vpc",
+		},
+		{
+			name: "PathWithRoot",
+			moduleURL: url.URL{
+				Scheme: "https",
+				Host:   "registry.terraform.io",
+				Path:   "modules/foo/bar",
+			},
+			terraformGet:   "/terraform-aws-modules/terraform-aws-vpc",
+			expectedResult: "https://registry.terraform.io/terraform-aws-modules/terraform-aws-vpc",
+		},
+		{
+			name: "PathWithRelativeRoot",
+			moduleURL: url.URL{
+				Scheme: "https",
+				Host:   "registry.terraform.io",
+				Path:   "modules/foo/bar",
+			},
+			terraformGet:   "./terraform-aws-modules/terraform-aws-vpc",
+			expectedResult: "https://registry.terraform.io/modules/foo/terraform-aws-modules/terraform-aws-vpc",
+		},
+		{
+			name: "PathWithRelativeParent",
+			moduleURL: url.URL{
+				Scheme: "https",
+				Host:   "registry.terraform.io",
+				Path:   "modules/foo/bar",
+			},
+			terraformGet:   "../terraform-aws-modules/terraform-aws-vpc",
+			expectedResult: "https://registry.terraform.io/modules/terraform-aws-modules/terraform-aws-vpc",
+		},
 	}
 
-	downloadURL, err := getDownloadURLFromHeader(context.Background(), testBaseURL, testTerraformGet)
-	require.NoError(t, err)
-	assert.Equal(t, "https://registry.terraform.io/terraform-aws-modules/terraform-aws-vpc", downloadURL)
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			downloadURL, err := getDownloadURLFromHeader(context.Background(), testCase.moduleURL, testCase.terraformGet)
```

Comment:
Make sense to move `t.Parallel()` from function beginning test case function in range loop?

Reply:
Since this is super fast (in memory), I wanted to avoid the complexity of using `t.Parallel` with the range variable. Introducing `t.Parallel` requires capturing `testCase` into the range variable, which requires some explanation (since it's confusing to see `testCase := testCase` to the uninitiated). I wanted to avoid all that explaining by not using it, since it's not necessary.

Hope that makes sense!